src/sophia/eventgen.f

```diff
@@ -286,6 +293,7 @@ subroutine eventgen(L0,E0,eps,theta,Imode)
          enddo
         endif
         NP = istable       
+       end if  ! anpros 2022/07/22  REMDEC condition end       
```

Comment:
There is a file called impy_config.yaml. There should be a definition of SOPHIA. The behavior of removing the decayed particles should be controlled from there. It's something that doesn't or shouldn't change in runtime. You can put the default to keep the particles, since most other models do it. The performance penalty for something fast as SOPHIA is totally negligible.

Reply:
The dictionary to sophia is added to impy_config.yaml. It mostly copies sibyll with relevant changes. However, the meaning of some of the entries is not clear to me, so this should be checked. keep_decayed_particles : True is added as default, to control keeping of decayed particles. Actually, in this case there is no performance penalty, because keeping particles do not require any operations as opposed to the removing.